File: src/theme/spacing.ts
/**
 * Zutsu-Log スペーシング・レイアウト
 * 間隔、角丸、シャドウの定義
 * ハードコード禁止 - 必ずこのファイルから参照すること
 */

export const spacing = {
  /** 2px */
  '2xs': 2,
  /** 4px */
  xs: 4,
  /** 6px */
  sm: 6,
  /** 8px */
  md: 8,
  /** 12px */
  lg: 12,
  /** 16px */
  xl: 16,
  /** 20px */
  '2xl': 20,
  /** 24px */
  '3xl': 24,
  /** 32px */
  '4xl': 32,
  /** 40px */
  '5xl': 40,
  /** 48px */
  '6xl': 48,
} as const;

export const borderRadius = {
  /** 4px - 小さなバッジ */
  sm: 4,
  /** 8px - ボタン、入力フィールド */
  md: 8,
  /** 12px - カード（iOS風） */
  lg: 12,
  /** 16px - 大きなカード */
  xl: 16,
  /** 20px - パネル */
  '2xl': 20,
  /** 24px - モーダル */
  '3xl': 24,
  /** 完全な円 */
  full: 9999,
} as const;

// iOS風のシャドウ（React Native用）
export const shadows = {
  none: {
    shadowColor: 'transparent',
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0,
    shadowRadius: 0,
    elevation: 0,
  },
  sm: {
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 0.05,
    shadowRadius: 2,
    elevation: 1,
  },
  md: {
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.08,
    shadowRadius: 4,
    elevation: 2,
  },
  lg: {
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 4,
  },
  xl: {
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 8 },
    shadowOpacity: 0.12,
    shadowRadius: 16,
    elevation: 8,
  },
  // プライマリカラーのシャドウ（ボタン用）
  primarySm: {
    shadowColor: '#4A90E2',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 4,
    elevation: 2,
  },
  primaryLg: {
    shadowColor: '#4A90E2',
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.3,
    shadowRadius: 8,
    elevation: 4,
  },
} as const;

// よく使うレイアウト値
export const layout = {
  // 画面のパディング
  screenPaddingX: spacing.xl,    // 16px
  screenPaddingY: spacing['2xl'], // 20px

  // カードのパディング
  cardPaddingX: spacing['2xl'],  // 20px
  cardPaddingY: spacing.xl,      // 16px

  // ヘッダーの高さ
  headerHeight: 56,

  // ボトムナビの高さ
  bottomNavHeight: 60,
  bottomNavHeightWithSafe: 80,

  // FABのサイズ
  fabSize: 56,

  // アイコンサイズ
  iconSm: 18,
  iconMd: 22,
  iconLg: 26,
  iconXl: 30,

  // アバター・サムネイル
  avatarSm: 32,
  avatarMd: 40,
  avatarLg: 48,

  // タイムラインのアイコン
  timelineIconSize: 40,

  // トグルスイッチ
  toggleWidth: 44,
  toggleHeight: 24,
  toggleThumb: 20,
} as const;

// 安全領域（Safe Area）
export const safeArea = {
  // iOS のノッチ対応
  top: 44,
  bottom: 34,
} as const;

export type Spacing = typeof spacing;
export type BorderRadius = typeof borderRadius;
export type Shadows = typeof shadows;

File: flutter/lib/theme/spacing.dart
import 'package:flutter/material.dart';

/// スペーシング定数
class Spacing {
  Spacing._();

  /// 2px
  static const double xxs = 2;

  /// 4px
  static const double xs = 4;

  /// 6px
  static const double sm = 6;

  /// 8px
  static const double md = 8;

  /// 12px
  static const double lg = 12;

  /// 16px
  static const double xl = 16;

  /// 20px
  static const double xxl = 20;

  /// 24px
  static const double xxxl = 24;

  /// 32px
  static const double xxxxl = 32;

  /// 40px
  static const double xxxxxl = 40;

  /// 48px
  static const double xxxxxxl = 48;
}

/// 角丸定数
class AppRadius {
  AppRadius._();

  /// 4px - 小さなバッジ
  static const double sm = 4;

  /// 8px - ボタン、入力フィールド
  static const double md = 8;

  /// 12px - カード（iOS風）
  static const double lg = 12;

  /// 16px - 大きなカード
  static const double xl = 16;

  /// 20px - パネル
  static const double xxl = 20;

  /// 24px - モーダル
  static const double xxxl = 24;

  /// BorderRadius
  static BorderRadius get smBorder => BorderRadius.circular(sm);
  static BorderRadius get mdBorder => BorderRadius.circular(md);
  static BorderRadius get lgBorder => BorderRadius.circular(lg);
  static BorderRadius get xlBorder => BorderRadius.circular(xl);
  static BorderRadius get xxlBorder => BorderRadius.circular(xxl);
  static BorderRadius get xxxlBorder => BorderRadius.circular(xxxl);
}

/// シャドウ定数
class AppShadows {
  AppShadows._();

  static List<BoxShadow> get none => [];

  static List<BoxShadow> get sm => [
        BoxShadow(
          color: Colors.black.withOpacity(0.05),
          offset: const Offset(0, 1),
          blurRadius: 2,
        ),
      ];

  static List<BoxShadow> get md => [
        BoxShadow(
          color: Colors.black.withOpacity(0.08),
          offset: const Offset(0, 2),
          blurRadius: 4,
        ),
      ];

  static List<BoxShadow> get lg => [
        BoxShadow(
          color: Colors.black.withOpacity(0.1),
          offset: const Offset(0, 4),
          blurRadius: 8,
        ),
      ];

  static List<BoxShadow> get xl => [
        BoxShadow(
          color: Colors.black.withOpacity(0.12),
          offset: const Offset(0, 8),
          blurRadius: 16,
        ),
      ];

  /// プライマリカラーのシャドウ（ボタン用）
  static List<BoxShadow> get primarySm => [
        BoxShadow(
          color: const Color(0xFF4A90E2).withOpacity(0.2),
          offset: const Offset(0, 2),
          blurRadius: 4,
        ),
      ];

  static List<BoxShadow> get primaryLg => [
        BoxShadow(
          color: const Color(0xFF4A90E2).withOpacity(0.3),
          offset: const Offset(0, 4),
          blurRadius: 8,
        ),
      ];
}

/// レイアウト定数
class Layout {
  Layout._();

  /// 画面のパディング
  static const double screenPaddingX = Spacing.xl; // 16px
  static const double screenPaddingY = Spacing.xxl; // 20px

  /// カードのパディング
  static const double cardPaddingX = Spacing.xxl; // 20px
  static const double cardPaddingY = Spacing.xl; // 16px

  /// ヘッダーの高さ
  static const double headerHeight = 56;

  /// ボトムナビの高さ
  static const double bottomNavHeight = 60;
  static const double bottomNavHeightWithSafe = 80;

  /// FABのサイズ
  static const double fabSize = 56;

  /// アイコンサイズ
  static const double iconSm = 18;
  static const double iconMd = 22;
  static const double iconLg = 26;
  static const double iconXl = 30;

  /// アバター・サムネイル
  static const double avatarSm = 32;
  static const double avatarMd = 40;
  static const double avatarLg = 48;

  /// タイムラインのアイコン
  static const double timelineIconSize = 40;

  /// トグルスイッチ
  static const double toggleWidth = 44;
  static const double toggleHeight = 24;
  static const double toggleThumb = 20;

  /// EdgeInsets ヘルパー
  static EdgeInsets get screenPadding =>
      const EdgeInsets.symmetric(horizontal: screenPaddingX, vertical: screenPaddingY);

  static EdgeInsets get cardPadding =>
      const EdgeInsets.symmetric(horizontal: cardPaddingX, vertical: cardPaddingY);
}
